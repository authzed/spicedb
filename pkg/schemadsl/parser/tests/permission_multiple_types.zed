definition mydefinition {
    relation viewer: user
    
    permission view: user | organization = viewer
    permission edit: user | organization | group = viewer
    permission admin: group = viewer
}