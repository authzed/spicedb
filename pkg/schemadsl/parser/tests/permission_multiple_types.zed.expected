NodeTypeFile
  end-rune = 202
  input-source = permission multiple types test
  start-rune = 0
  child-node =>
    NodeTypeDefinition
      definition-name = mydefinition
      end-rune = 202
      input-source = permission multiple types test
      start-rune = 0
      child-node =>
        NodeTypeRelation
          end-rune = 50
          input-source = permission multiple types test
          relation-name = viewer
          start-rune = 30
          allowed-types =>
            NodeTypeTypeReference
              end-rune = 50
              input-source = permission multiple types test
              start-rune = 47
              type-ref-type =>
                NodeTypeSpecificTypeReference
                  end-rune = 50
                  input-source = permission multiple types test
                  start-rune = 47
                  type-name = user
        NodeTypePermission
          end-rune = 105
          input-source = permission multiple types test
          relation-name = view
          start-rune = 61
          compute-expression =>
            NodeTypeIdentifier
              end-rune = 105
              identifier-value = viewer
              input-source = permission multiple types test
              start-rune = 100
          type-annotations =>
            NodeTypeTypeAnnotation
              end-rune = 96
              input-source = permission multiple types test
              start-rune = 78
              annotation-types =>
                NodeTypeIdentifier
                  end-rune = 81
                  identifier-value = user
                  input-source = permission multiple types test
                  start-rune = 83
                NodeTypeIdentifier
                  end-rune = 96
                  identifier-value = organization
                  input-source = permission multiple types test
                  start-rune = 98
        NodeTypePermission
          end-rune = 163
          input-source = permission multiple types test
          relation-name = edit
          start-rune = 111
          compute-expression =>
            NodeTypeIdentifier
              end-rune = 163
              identifier-value = viewer
              input-source = permission multiple types test
              start-rune = 158
          type-annotations =>
            NodeTypeTypeAnnotation
              end-rune = 154
              input-source = permission multiple types test
              start-rune = 128
              annotation-types =>
                NodeTypeIdentifier
                  end-rune = 131
                  identifier-value = user
                  input-source = permission multiple types test
                  start-rune = 133
                NodeTypeIdentifier
                  end-rune = 146
                  identifier-value = organization
                  input-source = permission multiple types test
                  start-rune = 148
                NodeTypeIdentifier
                  end-rune = 154
                  identifier-value = group
                  input-source = permission multiple types test
                  start-rune = 156
        NodeTypePermission
          end-rune = 200
          input-source = permission multiple types test
          relation-name = admin
          start-rune = 169
          compute-expression =>
            NodeTypeIdentifier
              end-rune = 200
              identifier-value = viewer
              input-source = permission multiple types test
              start-rune = 195
          type-annotations =>
            NodeTypeTypeAnnotation
              end-rune = 191
              input-source = permission multiple types test
              start-rune = 187
              annotation-types =>
                NodeTypeIdentifier
                  end-rune = 191
                  identifier-value = group
                  input-source = permission multiple types test
                  start-rune = 193